src/Interpreters/Cache/FileCache.cpp

```diff
@@ -203,8 +240,6 @@ void FileCache::initialize()
         keep_up_free_space_ratio_task = Context::getGlobalContextInstance()->getSchedulePool().createTask(log->name(), [this] { freeSpaceRatioKeepingThreadFunc(); });
         keep_up_free_space_ratio_task->schedule();
     }
-
-    is_initialized = true;
```

Comment:
I'd rather leave is_initalized setting to true here, as it was before, now you moved to loadMetadataImpl(), which could lead to unnoticed error from this part of the code, while it is very important

Reply:
Changed the location of the assignment